### DISCUSSION TITLE
compatibility with python 3.11.1 and QT Designer

### DISCUSSION THREAD

**Discussion by igutiez** [gijzelaerr/python-snap7#417] [Q&A] [answered]
I have developed some custom plugins for Qt designer for connecting with Siemens PLC. They are working pretty well using 3.10.x or 3.11.0 but there are something in the last 3.11.1 that is not compatible with Qt-Designer + snap7. I am sure that the problem is with snap7 because if I not import the snap7 package, the plugins appear again in Qt-Designer. 
It is something related to 3.11.1 - Snap7, but I cannot find the problem.

**Comment by gijzelaerr** [comment, marked as answer]
i have no problem installing snap7 on python 3.11.1:
```
λ  python3.11 --version
Python 3.11.1
λ  python3.11 -m venv venv
λ  venv/bin/pip install python-snap7
Looking in indexes: https://artifactory.spotify.net/artifactory/api/pypi/pypi/simple/, https://artifactory.spotify.net/artifactory/api/pypi/pypi/simple/
Collecting python-snap7
  Downloading https://artifactory.spotify.net/artifactory/api/pypi/pypi/packages/packages/b6/d3/d97b2d85f31a7a6b63231e6090836914cb8bec91060a8b2b619c4e20c2d8/python_snap7-1.2-py3-none-macosx_10_9_universal2.whl (192 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 192.1/192.1 kB 220.9 kB/s eta 0:00:00
Installing collected packages: python-snap7
Successfully installed python-snap7-1.2
λ  venv/bin/python3 -c "import snap7"
```
So this is probably a problem with your setup or QT Designer. Therefore, converting this to a discussion.

**Reply by igutiez** [reply]
I have tested with the same project python+Qt designer+snap7:
Windows 10/11 + Python 3.10.9 /3.11.0 +Snap 1.2: works 
Windows 10/11 + Python 3.11.1 + Snap 1.2: not works

When I say that it doesnt work I mean that Qt-Designer doesnt loads the plugins when you want to work with designer.
The communication works well in all the versions with my projects, but I cannot edit in Qt-Designer using my plugins.

In my plugins, I have deleted the references to snap7 and it works also in python 3.11.1
I have programmed an empty plugin, just with an "import snap7" with anything else, and then, the plugin disapears when I open Qt-Designer.
I have try with other references, importing numpy and cv2 for example, and the plugin loads. Then I add just an "import snap7" and the plugin is not loaded.

It is just add "import snap7" and it stops working, with no aditional reference to snap7, funcion or whatever, just adding the "import snap7". Of course, snap7 is installed in the machine.

Also If I try to add from snap7 import util or client and so on, the problem is the same.

I have tested this situation in two different machines with Windows 11, and in two VM with Windows 10. The situation in all of the is the same.

**Reply by igutiez** [reply]
I have recorded a video with this issue.
https://vimeo.com/789963837
With 3.11.0 the shown issue is not present.

**Reply by gijzelaerr** [reply]
did you check the QT designer debug/error logs?

**Reply by igutiez** [reply]
I have searched, but I have not found any log.
The only thing there is in the "about" the loaded plugins, and it says nothing. When it loads the plugins the logs shows that the plugins are loaded, and when it doesn't load it doesn't say anything.
I dont know it it helps, but when I have tried to make an exe, snap7 is not included well. I have to "collect_all" of snap7 using pyinstaller to make the exe works.
It happens even in previous versions. I added the collect_all to pyinstaller, and snap7 is included properly. 
As I'm saying I dont know if there is any connection between both issues.

**Reply by igutiez** [reply]
I am making some test, and I think there is any problem with ctypes (byref, create_string_buffer, sizeof...) used in snap7.
So the problem is Qt-Designer with this package, so there is nothing to do for a while.
I have just added import ctypes, and the plugins are not loaded.

**Reply by gijzelaerr** [reply]
it is hard to tell without any debug logging from QT designer about what actually is going wrong.

**Reply by igutiez** [reply]
Yep. There is no log when Qt-Designer starts (or at least I don't know where to find it).
I whish it is solved in following releases of qt-desiger/python.
Thanks for your support.

**Reply by gijzelaerr** [reply]
maybe you can build QT designer yourself with debug flags and examine the logs?
